engine/src/org/pentaho/di/trans/steps/databaselookup/DatabaseLookup.java

```diff
@@ -77,7 +71,8 @@ public DatabaseLookup( StepMeta stepMeta, StepDataInterface stepDataInterface, i
    * @return the resulting row after the lookup values where added
    * @throws KettleException In case something goes wrong.
    */
-  private synchronized Object[] lookupValues( RowMetaInterface inputRowMeta, Object[] row ) throws KettleException {
+  // package-local visibility for testing purposes
```

Comment:
@akhayrutdinov I spoke with Kincade and asked around and the general consensus is if guava is already a dependency, @VisibleForTesting is preferred